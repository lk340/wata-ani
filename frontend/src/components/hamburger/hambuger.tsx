import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./hamburger.styled";
import * as Springs from "./hamburger.springs";

export const Hamburger = () => {
	const { navbar } = Context.Navbar.useNavbarContext();

	const animateLineTop = Springs.lineTop(navbar.state.hamburgerOpen);
	const animateLineMiddleOne = Springs.lineMiddle(navbar.state.hamburgerOpen, 45);
	const animateLineMiddleTwo = Springs.lineMiddle(navbar.state.hamburgerOpen, -45);
	const animateLineBottom = Springs.lineBottom(navbar.state.hamburgerOpen);

	return (
		<Styled.Hamburger onClick={navbar.setters.toggleHamburgerOpen}>
			<Styled.HamburgerLines>
				<Styled.HamburgerLinesLine style={animateLineTop} />
				<Styled.HamburgerLinesLineMiddle style={animateLineMiddleOne} />
				<Styled.HamburgerLinesLineMiddle style={animateLineMiddleTwo} />
				<Styled.HamburgerLinesLine style={animateLineBottom} />
			</Styled.HamburgerLines>
		</Styled.Hamburger>
	);
};
