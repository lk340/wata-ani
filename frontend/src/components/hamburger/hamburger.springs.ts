import * as Spring from "react-spring";

export function lineTop(open: boolean) {
	return Spring.useSpring({
		from: { top: "0%", transform: "translateY(0%)", opacity: "1" },
		to: open
			? { top: "50%", transform: "translateY(-50%)", opacity: "0" }
			: { top: "0%", transform: "translateY(0%)", opacity: "1" },
		config: { duration: 150 },
		delay: open ? 0 : 200,
	});
}

export function lineMiddle(open: boolean, degrees: number) {
	return Spring.useSpring({
		from: { transform: "translateY(-50%) rotate(0deg)" },
		to: open
			? {
					transform: `translateY(-50%) rotate(${degrees}deg)`,
			  }
			: {
					transform: "translateY(-50%) rotate(0deg)",
			  },
		config: { duration: 150 },
		delay: open ? 200 : 0,
	});
}

export function lineBottom(open: boolean) {
	return Spring.useSpring({
		from: { bottom: "0%", transform: "translateY(0%)", opacity: "1" },
		to: open
			? { bottom: "50%", transform: "translateY(50%)", opacity: "0" }
			: { bottom: "0%", transform: "translateY(0%)", opacity: "1" },
		config: { duration: 150 },
		delay: open ? 0 : 200,
	});
}
