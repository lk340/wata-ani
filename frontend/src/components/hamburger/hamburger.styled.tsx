import styled, { css } from "styled-components";
import { animated } from "react-spring";
import * as Gatsby from "gatsby";

import * as Constants from "@/utils/style/constants";
import * as Snippets from "@/utils/style/snippets";
import * as Colors from "@/utils/style/colors";

import home from "@/icons/mobile-modal/home.svg";
import search from "@/icons/mobile-modal/search.svg";
import settings from "@/icons/mobile-modal/settings.svg";
import registration from "@/icons/mobile-modal/registration.svg";
import signIn from "@/icons/mobile-modal/sign-in.svg";

/**
 * Hamburger
 * Lines
 * Modal
 * Modal Option
 */

// ================= //
// ↓↓↓ Hamburger ↓↓↓ //
// ================= //

export const Hamburger = styled("div")`
	${Snippets.flexColumnCenter()};
	${Snippets.square("20px")};
	display: none;

	@media (max-width: 575px) {
		display: flex;
	}
`;

// ============= //
// ↓↓↓ Lines ↓↓↓ //
// ============= //

export const HamburgerLines = styled("div")`
	${Snippets.fillContainer()};
	position: relative;
	z-index: 3;
	cursor: pointer;
`;

const lineStyles = css`
	${Snippets.size("100%", "4px")};
	position: absolute;
	background-color: ${Colors.PRIMARY_100};
	border-radius: 10rem;
`;

export const HamburgerLinesLine = styled(animated.div)`
	${lineStyles};
`;

export const HamburgerLinesLineMiddle = styled(animated.div)`
	${lineStyles};
	top: 50%;
	transform: translateY(-50%);
`;

// ============= //
// ↓↓↓ Modal ↓↓↓ //
// ============= //

export const HamburgerModal = styled(animated.div)`
	${Snippets.absolute("0px", "0px", "0px", "0px")};
	background-color: ${Colors.PRIMARY_100};
`;

export const HamburgerModalMain = styled(animated.div)`
	${Snippets.absolute("0px", "0px", "0px", "0px", 2)};
	${Snippets.fillContainer()};
`;

// ==================== //
// ↓↓↓ Modal Option ↓↓↓ //
// ==================== //

export const HamburgerModalMainOptionContainer = styled("div")`
	${Snippets.grid(1, "auto", 40)};
	margin-top: 90px;
`;

export const HamburgerModalMainOption = styled("div")`
	display: flex;
`;

export type OptionIcon = {
	iconType: "home" | "search" | "settings" | "sign in" | "registration";
};

export const HamburgerModalMainOptionIcon = styled("img").attrs((props: OptionIcon) => ({
	src:
		props.iconType === "home"
			? home
			: props.iconType === "search"
			? search
			: props.iconType === "settings"
			? settings
			: props.iconType === "registration"
			? registration
			: signIn,
	alt: `hamburger modal ${props.iconType} icon`,
}))<OptionIcon>`
	${Snippets.square("20px")};
	margin-right: 20px;
`;

export const HamburgerModalMainOptionButton = styled("div")`
	cursor: pointer;
`;

type OptionLink = { primary: boolean };

export const HamburgerModalMainOptionLink = styled(Gatsby.Link)<OptionLink>`
	${Snippets.clearAnchor()};
	color: ${(props) => (props.primary ? Colors.PRIMARY_100 : "inherit")};
	font-size: ${Constants.fontSizes.components.hamburger.link};
`;
