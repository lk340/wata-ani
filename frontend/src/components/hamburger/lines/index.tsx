import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./lines.styled";
import * as Springs from "./lines.springs";

export const Lines = () => {
	const { navbar } = Context.Navbar.useNavbarContext();

	const animateLineTop = Springs.lineTop(navbar.state.hamburgerOpen);
	const animateLineMiddleOne = Springs.lineMiddle(navbar.state.hamburgerOpen, 45);
	const animateLineMiddleTwo = Springs.lineMiddle(navbar.state.hamburgerOpen, -45);
	const animateLineBottom = Springs.lineBottom(navbar.state.hamburgerOpen);

	return (
		<Styled.Lines onClick={navbar.setters.toggleHamburgerOpen}>
			<Styled.LinesLine style={animateLineTop} />
			<Styled.LinesLineMiddle style={animateLineMiddleOne} />
			<Styled.LinesLineMiddle style={animateLineMiddleTwo} />
			<Styled.LinesLine style={animateLineBottom} />
		</Styled.Lines>
	);
};
