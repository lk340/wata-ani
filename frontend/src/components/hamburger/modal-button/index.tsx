import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./modal-button.styled";

import { OptionProps } from "../modal-options";

type Props = {
	buttonType: "search" | "settings";
	test_id: string;
} & OptionProps;

export const ModalButton = (props: Props) => {
	const { iconType, text, display, buttonType, test_id } = props;

	const { navbar } = Context.Navbar.useNavbarContext();

	return (
		<Styled.ModalButton display={display.toString()} test_id={test_id}>
			<Styled.ModalButtonIcon iconType={iconType} />
			<Styled.ModalButtonText
				onClick={
					buttonType === "search"
						? navbar.setters.toggleSearch
						: navbar.setters.toggleSettings
				}
			>
				{text}
			</Styled.ModalButtonText>
		</Styled.ModalButton>
	);
};
