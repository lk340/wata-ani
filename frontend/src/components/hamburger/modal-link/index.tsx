import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./modal-link.styled";

import { OptionProps } from "../modal-options";

type Props = {
	to: string;
	primary: boolean;
	test_id: string;
} & OptionProps;

export const ModalLink = (props: Props) => {
	const { iconType, text, display, to, primary, test_id } = props;

	const { navbar } = Context.Navbar.useNavbarContext();

	return (
		<Styled.ModalLink display={display.toString()} test_id={test_id}>
			<Styled.ModalLinkIcon iconType={iconType} />
			<Styled.ModalLinkText
				to={to}
				onClick={navbar.setters.toggleHamburgerOpen}
				primary={primary.toString()}
			>
				{text}
			</Styled.ModalLinkText>
		</Styled.ModalLink>
	);
};
