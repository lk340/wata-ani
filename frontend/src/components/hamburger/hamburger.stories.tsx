import * as React from "react";
import * as Reach from "@reach/router";

import * as Wrappers from "@/wrappers";

import { Hamburger } from "./hamburger";

type Story = {
	component: React.ReactNode;
	title: string;
	excludeStories: RegExp;
};

const story: Story = {
	component: Hamburger,
	title: "Hamburger",
	excludeStories: /.*Data$/,
};

export default story;

export const hamburgerData = {};

export const Default = () => {
	return (
		<Reach.LocationProvider>
			<Wrappers.Layout>
				<Hamburger />
			</Wrappers.Layout>
		</Reach.LocationProvider>
	);
};
