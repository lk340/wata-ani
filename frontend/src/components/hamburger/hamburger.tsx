import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./hamburger.styled";
import * as Springs from "./hamburger.springs";

export const Hamburger = () => {
	const { navbar } = Context.Navbar.useNavbarContext();
	const { theme } = Context.Theme.useThemeContext();

	// --- Line Animations --- //
	const animateLineTop = Springs.lineTop(navbar.state.hamburgerOpen);
	const animateLineMiddleOne = Springs.lineMiddle(navbar.state.hamburgerOpen, 45);
	const animateLineMiddleTwo = Springs.lineMiddle(navbar.state.hamburgerOpen, -45);
	const animateLineBottom = Springs.lineBottom(navbar.state.hamburgerOpen);

	// --- Modal Animations --- //
	const animateModal = Springs.modal(navbar.state.hamburgerOpen);
	const animateModalMain = Springs.modalMain(
		navbar.state.hamburgerOpen,
		theme.state.mode,
	);

	return (
		<Styled.Hamburger>
			{/* Lines */}
			<Styled.HamburgerLines onClick={navbar.setters.toggleHamburgerOpen}>
				<Styled.HamburgerLinesLine style={animateLineTop} />
				<Styled.HamburgerLinesLineMiddle style={animateLineMiddleOne} />
				<Styled.HamburgerLinesLineMiddle style={animateLineMiddleTwo} />
				<Styled.HamburgerLinesLine style={animateLineBottom} />
			</Styled.HamburgerLines>

			{/* Modal */}
			<Styled.HamburgerModal style={animateModal}>
				<Styled.HamburgerModalMain style={animateModalMain}>
					<Options />
				</Styled.HamburgerModalMain>
			</Styled.HamburgerModal>
		</Styled.Hamburger>
	);
};

// ======================== //
// ↓↓↓ Options ↓↓↓ //
// ======================== //

const Options = () => {
	return (
		<Styled.HamburgerModalMainOptionContainer>
			{/* Home */}
			<OptionLink iconType="home" text="Home" display={true} to="/" primary={false} />
			{/* Search */}
			<OptionButton iconType="search" text="Search" display={true} option="search" />
			{/* Sign In */}
			<OptionLink
				iconType="sign in"
				text="Sign In"
				display={true}
				to="/sign-in"
				primary={false}
			/>
			{/* Registration */}
			<OptionLink
				iconType="registration"
				text="Register"
				display={true}
				to="/registration"
				primary={true}
			/>
			{/* Settings */}
			<OptionButton
				iconType="settings"
				text="Settings"
				display={true}
				option="settings"
			/>
		</Styled.HamburgerModalMainOptionContainer>
	);
};

// ============== //
// ↓↓↓ Option ↓↓↓ //
// ============== //

type OptionProps = {
	iconType: "home" | "search" | "settings" | "sign in" | "registration";
	text: string;
	display: boolean;
};

type OptionButtonProps = { option: Context.Navbar.Options } & OptionProps;

const OptionButton = (props: OptionButtonProps) => {
	const { iconType, text, display, option } = props;

	const { navbar } = Context.Navbar.useNavbarContext();

	return (
		<Styled.HamburgerModalMainOption display={display.toString()}>
			<Styled.HamburgerModalMainOptionIcon iconType={iconType} />
			<Styled.HamburgerModalMainOptionButton
				onClick={
					option === "search"
						? navbar.setters.toggleSearch
						: navbar.setters.toggleSettings
				}
			>
				{text}
			</Styled.HamburgerModalMainOptionButton>
		</Styled.HamburgerModalMainOption>
	);
};

type OptionLinkProps = {
	to: string;
	primary: boolean;
} & OptionProps;

const OptionLink = (props: OptionLinkProps) => {
	const { iconType, text, display, to, primary } = props;

	const { navbar } = Context.Navbar.useNavbarContext();

	return (
		<Styled.HamburgerModalMainOption display={display.toString()}>
			<Styled.HamburgerModalMainOptionIcon iconType={iconType} />
			<Styled.HamburgerModalMainOptionLink
				to={to}
				onClick={navbar.setters.toggleHamburgerOpen}
				primary={primary.toString()}
			>
				{text}
			</Styled.HamburgerModalMainOptionLink>
		</Styled.HamburgerModalMainOption>
	);
};
