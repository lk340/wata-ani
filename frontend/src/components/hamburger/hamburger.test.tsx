import * as React from "react";
import * as Testing from "@testing-library/react";
import axios from "axios";

import * as Context from "@/context";
import * as Tests from "@/utils/tests";

import { Hamburger } from "./hamburger";

jest.mock("axios");

describe("Hamburger Component", () => {
	it("matches the snapshot", () => {
		const { asFragment } = Tests.renderWithContext(<Hamburger />);

		expect(asFragment(<Hamburger />)).toMatchSnapshot();
	});

	it("opens and closes based on state", () => {
		const HamburgerWithModalState = () => {
			const { navbar } = Context.Navbar.useNavbarContext();

			return (
				<React.Fragment>
					<Hamburger />
					<div data-testid="modalState">{navbar.state.hamburgerOpen.toString()}</div>
				</React.Fragment>
			);
		};

		const hamburgerWithTheme = Tests.renderWithContext(<HamburgerWithModalState />);

		const lines = hamburgerWithTheme.getByTestId(/hamburger component lines/);
		expect(lines).not.toBe(null);

		const hamburgerOpen = hamburgerWithTheme.getByTestId(/modalState/);
		expect(hamburgerOpen).not.toBe(null);
		expect(hamburgerOpen).toHaveTextContent(/false/);

		Testing.fireEvent.click(lines);
		expect(hamburgerOpen).toHaveTextContent(/true/);

		Testing.fireEvent.click(lines);
		expect(hamburgerOpen).toHaveTextContent(/false/);
	});

	it("displays the Home link, Sign In link, and Register link when user is not signed in", () => {
		const hamburger = Tests.renderWithContext(<Hamburger />);

		const homeLink = hamburger.getByTestId(/hamburger component home link/);
		const searchButton = hamburger.getByTestId(/hamburger component search button/);
		const signInLink = hamburger.getByTestId(/hamburger component sign in link/);
		const SignOutButton = hamburger.getByTestId(/hamburger component sign out button/);
		const registerLink = hamburger.getByTestId(/hamburger component registration link/);
		const settingsButton = hamburger.getByTestId(/hamburger component settings button/);

		expect(homeLink).not.toBe(null);
		expect(homeLink).toHaveStyle("display: flex");

		expect(searchButton).not.toBe(null);
		expect(searchButton).toHaveStyle("display: none");

		expect(signInLink).not.toBe(null);
		expect(signInLink).toHaveStyle("display: flex");

		expect(SignOutButton).not.toBe(null);
		expect(SignOutButton).toHaveStyle("display: none");

		expect(registerLink).not.toBe(null);
		expect(registerLink).toHaveStyle("display: flex");

		expect(settingsButton).not.toBe(null);
		expect(settingsButton).toHaveStyle("display: none");
	});

	it("displays the Home link, Search button, Sign Out button, and Settings button when user is signed in", async () => {
		const registrationData = {
			username: "foobar",
			email: "foo@bar.com",
			password1: "newyorkstateofmind",
			password2: "newyorkstateofmind",
		};

		await axios.post("/auth/registration/", registrationData);

		const signInData = { username: "foobar", password: "newyorkstateofmind" };

		await axios.post("/auth/login/", signInData);

		const hamburger = Tests.renderWithContext(<Hamburger />);

		const homeLink = hamburger.getByTestId(/hamburger component home link/);
		const searchButton = hamburger.getByTestId(/hamburger component search button/);
		const signInLink = hamburger.getByTestId(/hamburger component sign in link/);
		const SignOutButton = hamburger.getByTestId(/hamburger component sign out button/);
		const registerLink = hamburger.getByTestId(/hamburger component registration link/);
		const settingsButton = hamburger.getByTestId(/hamburger component settings button/);

		expect(homeLink).not.toBe(null);
		expect(homeLink).toHaveStyle("display: flex");

		expect(searchButton).not.toBe(null);
		expect(searchButton).toHaveStyle("display: flex");

		expect(signInLink).not.toBe(null);
		expect(signInLink).toHaveStyle("display: none");

		expect(SignOutButton).not.toBe(null);
		expect(SignOutButton).toHaveStyle("display: flex");

		expect(registerLink).not.toBe(null);
		expect(registerLink).toHaveStyle("display: none");

		expect(settingsButton).not.toBe(null);
		expect(settingsButton).toHaveStyle("display: flex");
	});
});
