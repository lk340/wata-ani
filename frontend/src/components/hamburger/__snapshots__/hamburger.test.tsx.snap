// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Component matches the snapshot 1`] = `
<DocumentFragment>
  .c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  margin: 0px 40px;
  cursor: pointer;
}

.c20 {
  width: 20px;
  height: 20px;
  position: absolute;
}

.c21 {
  width: 20px;
  height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  display: none;
}

.c4 {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 1;
  background-color: #6BA6FF;
}

.c5 {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 2;
  width: 100%;
  height: 100%;
}

.c6 {
  width: 100%;
  height: 100%;
}

.c18 {
  position: absolute;
  top: auto;
  right: 0px;
  bottom: 30px;
  left: auto;
  z-index: 3;
}

.c1 {
  width: 100%;
  height: 100%;
  position: relative;
  z-index: 3;
  cursor: pointer;
}

.c2 {
  width: 100%;
  height: 4px;
  position: absolute;
  background-color: #6BA6FF;
  border-radius: 10rem;
}

.c3 {
  width: 100%;
  height: 4px;
  position: absolute;
  background-color: #6BA6FF;
  border-radius: 10rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c7 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 40px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
  position: absolute;
  right: 0;
  margin-top: 90px;
  padding-right: 30px;
}

.c11 {
  display: none;
}

.c12 {
  width: 20px;
  height: 20px;
  margin-right: 20px;
}

.c13 {
  display: block;
  cursor: pointer;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  width: 20px;
  height: 20px;
  margin-right: 20px;
}

.c10 {
  display: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  color: inherit;
  font-size: 0.875rem;
}

.c17 {
  display: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  color: #6BA6FF;
  font-size: 0.875rem;
}

.c14 {
  display: none;
}

.c15 {
  width: 20px;
  height: 20px;
  margin-right: 20px;
}

.c16 {
  display: block;
  cursor: pointer;
}

@media (max-width:575px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="hamburger component lines"
    >
      <div
        class="c2"
        style="top: 0%; transform: translateY(0%); opacity: 1;"
      />
      <div
        class="c3"
        style="transform: translateY(-50%) rotate(0deg);"
      />
      <div
        class="c3"
        style="transform: translateY(-50%) rotate(0deg);"
      />
      <div
        class="c2"
        style="bottom: 0%; transform: translateY(0%); opacity: 1;"
      />
    </div>
    <div
      class="c4"
      style="transform: translateX(100%);"
    >
      <div
        class="c5"
        style="transform: translateX(100%);"
      >
        <div
          class="c6"
          style="background-color: rgb(255, 255, 255);"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
              data-testid="hamburger component home link"
              display="true"
            >
              <img
                alt="hamburger modal home icon"
                class="c9"
                src="[object Object]"
              />
              <a
                class="c10"
                href="/"
                primary="false"
              >
                Home
              </a>
            </div>
            <div
              class="c11"
              data-testid="hamburger component search button"
              display="false"
            >
              <img
                alt="hamburger modal search icon"
                class="c12"
                src="[object Object]"
              />
              <span
                class="c13"
              >
                Search
              </span>
            </div>
            <div
              class="c8"
              data-testid="hamburger component sign in link"
              display="true"
            >
              <img
                alt="hamburger modal sign in icon"
                class="c9"
                src="[object Object]"
              />
              <a
                class="c10"
                href="/sign-in"
                primary="false"
              >
                Sign In
              </a>
            </div>
            <div
              class="c14"
              data-testid="hamburger component sign out button"
              display="false"
            >
              <img
                alt="hamburger modal sign out icon"
                class="c15"
                src="[object Object]"
              />
              <span
                class="c16"
              >
                Sign Out
              </span>
            </div>
            <div
              class="c8"
              data-testid="hamburger component registration link"
              display="true"
            >
              <img
                alt="hamburger modal registration icon"
                class="c9"
                src="[object Object]"
              />
              <a
                class="c17"
                href="/registration"
                primary="true"
              >
                Register
              </a>
            </div>
            <div
              class="c11"
              data-testid="hamburger component settings button"
              display="false"
            >
              <img
                alt="hamburger modal settings icon"
                class="c12"
                src="[object Object]"
              />
              <span
                class="c13"
              >
                Settings
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c18"
      >
        <div
          class="c19"
          data-testid="theme button component"
        >
          <img
            alt="navbar component light theme button"
            class="c20"
            src="[object Object]"
            style="opacity: 0;"
          />
          <img
            alt="navbar component dark theme button"
            class="c21"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
